coord/src/config.h

```diff
@@ -33,8 +34,9 @@ extern SearchClusterConfig clusterConfig;
     .numPartitions = 0,                                                                    \
     .connPerShard = 0,                                                                     \
     .type = DetectClusterType(),                                                           \
-    .timeoutMS = 500,                                                                      \
+    .timeoutMS = 0,                                                                        \
```

Comment:
Why changing default timeout? What am I missing?

Reply:
TL;DR: it was never in use, and the actual default value was 0, so this is for keeping the default value that was currently used

This default definition was never used, and we always set all the configurations to 0 by default. I noticed that when I made the new configuration with the default value of 1, but in tests it always got it set to 0.
We need to discuss this configuration. It has a few false default values in the code that are always ignored or overridden, and set to 0. The configuration name for the `FT.CONFIG` API is `TIMEOUT`, which is also the name of the default timeout of a query, making it inaccessible for modification (since the first `TIMEOUT` config that is found is always the query timeout one). It is also marked as available to modify on runtime but its value is copied to a global variable once at modul-init and is never changed again. Lastly, it is used for giving a timeout for the client blocking in the `rmr` calls. Before making this value 0, some tests started failing on random timeouts, and I'm not sure we want to use RM API for handling timeout like that. Maybe it was an initial idea for how to ensure that queries are finished after some time even if some shards did not reply on time.

Reply:
OK, lets just mention it on the PR top comment with detailed explanation on why it was broken and why the changes keeps the current broken behavior of timeout 0.